sw/device/lib/testing/clkmgr_testutils.c

```diff
@@ -27,27 +28,60 @@ typedef struct expected_count_info {
   uint32_t variability;
 } expected_count_info_t;
 
-// The expected counts are derived from the ratios of the frequencies of the
-// various clocks to the AON clock. For example, 48 Mhz / 200 kHz = 240, so
-// we set count to 479 and variability to 1, meaning the max threshold is 480,
-// and the min to 478. Notice the target counts could be computed from the
-// device frequencies, but they are expressed as uint64_t, and are not
-// compile-time constants.
-static const expected_count_info_t kNoJitterCountInfos[] = {
-    {.count = 479, .variability = 1}, {.count = 239, .variability = 1},
-    {.count = 119, .variability = 1}, {.count = 499, .variability = 1},
-    {.count = 239, .variability = 1},
-};
-
-// If jitter is enabled the low threshold should be up to 20% lower, so
-// the variability is set to 0.1 * max_count, and count as max - 0.1 * max.
-static const expected_count_info_t kJitterCountInfos[] = {
-    {.count = 480 - 48, .variability = 48},
-    {.count = 240 - 24, .variability = 24},
-    {.count = 120 - 12, .variability = 12},
-    {.count = 500 - 50, .variability = 50},
-    {.count = 240 - 24, .variability = 24},
-};
+// The expected counts when jitter is disabled.
+static expected_count_info_t kNoJitterCountInfos[kDifClkmgrMeasureClockUsb + 1];
+
+// The expected counts when jitter is enabled.
+static expected_count_info_t kJitterCountInfos[kDifClkmgrMeasureClockUsb + 1];
+
+static void initialize_expected_counts() {
+  // The expected counts depend on the device, per sw/device/lib/arch/device.h.
+  // Notice the ratios are small enough to fit a uint32_t, even if the Hz number
```

Comment:
Do we want to add some check to make sure thats the case. It holds now, but any device.h files were ever changed this might become a silent error no?

Reply:
Sure. Fixed.